assimilation_code/modules/io/state_structure_mod.f90

```diff
@@ -936,7 +947,76 @@ subroutine load_common_cf_conventions(domain)
 
 end subroutine load_common_cf_conventions
 
+!-------------------------------------------------------------------------------
+!> TIEGCM, the top level of each variable is the boundary condition.
+!> So the top level should not be part of the state.
+!> How general does this need to be?  At the moment it is just a hack to
+!> be able to proceed with TIEGCM.
+!>
+!> This slicing needs to be on the the DART state structure, not the io netcdf
+!> structure.
+
+subroutine hyperslice_domain(dom_id, dim_name, new_length)
+
+integer,          intent(in) :: dom_id
+character(len=*), intent(in) :: dim_name
+integer,          intent(in) :: new_length
+
+integer :: ivar, jdim
+integer(i8) :: orig_domain_size, orig_model_size, index_start
+integer(i8) :: domain_size, variable_size
+
+orig_domain_size = get_domain_size(dom_id)
+orig_model_size = state%model_size
+! Start of the domain remains the same
+index_start = state%domain(dom_id)%variable(1)%index_start
 
+! Reduce the dimension size
+do ivar = 1, get_num_variables(dom_id)
+   do jdim = 1, get_num_dims(dom_id, ivar)
+       if (get_dim_name(dom_id, ivar, jdim) == dim_name) then
+          state%domain(dom_id)%variable(ivar)%dimlens(jdim) = new_length
+       endif
+   enddo
+enddo
+
+! need to modify state index accessing for each variable:
+!   variable_size
+!   index_start, index_end
+!   model size
+!   domain size
+
+domain_size = 0
+! variable size
+do ivar = 1, get_num_variables(dom_id)
+
+   variable_size = 1
+
+   do jdim = 1, get_num_dims(dom_id, ivar)
+      !HK there is a todo about this, what is going on with time and member?
+      if ((state%domain(dom_id)%variable(ivar)%dimname(jdim) == 'time') .or. &
```

Comment:
i remember this being added to the existing code but i don't remember why, sorry.  "member" is the dimension created by dart when making a unified (single) file that holds all the ensemble member info.  "time" is most often an unlimited dim but nothing in netcdf forces it to be.